python/vyos/template.py

```diff
@@ -871,10 +871,74 @@ def kea_high_availability_json(config):
 
     return dumps(data)
 
+@register_filter('kea_dynamic_dns_update_main_json')
+def kea_dynamic_dns_update_main_json(config):
+    from vyos.kea import kea_parse_ddns_settings
+    from json import dumps
+
+    data = kea_parse_ddns_settings(config)
+
+    return dumps(data, indent=8)[1:-1] + ','
```

Comment:
This feels rather fragile. Do you think it's possible to rework the logic so that the JSON in question is produced from a dict at once, rather than by glueing chunks together?